lib/pbench/test/functional/server/test_connect.py

```diff
@@ -13,9 +13,14 @@ def test_connect(self, server_client: PbenchServerClient):
         endpoints = server_client.endpoints
         assert endpoints
         assert "api" in endpoints
+        assert "authentication" in endpoints
         assert "identification" in endpoints
         assert "uri" in endpoints
 
+        # verify all the required authentication fields are present
+        auth_fields = ["issuer", "client", "realm"]
+        assert all(e in endpoints["authentication"].keys() for e in auth_fields)
```

Comment:
Isn't this just the same as:

```
auth_fields = {"issuer", "client", "realm"}
keys = set(endpoints["authentication"].keys())
assert keys == auth_fields
```

Reply:
Ah! actually no, the `endpoints["authentication"]` could return `secret` which is an optional field. so they may not be similar always.

Reply:
Also, we don't care if `endpoints["authentication"]` return more fields we only want to make sure all the required fields are present in the JSON.

Reply:
Oh, right.  (The saliva got in my eyes....  😆)  How about
```
        assert set(endpoints["authentication"]) >= {"issuer", "client", "realm"}
```

Reply:
Fixed.